packages/core/src/external/cda/process-attachments.ts

```diff
@@ -79,26 +88,47 @@ export async function processAttachments({
       mimeType: fileDetails.mimeType,
     });
     const fileUrl = s3Utils.buildFileUrl(s3BucketName, fileKey);
-    const attachment: Attachment = {
-      ...(fileDetails.mimeType && { contentType: fileDetails.mimeType }),
-      url: fileUrl,
-      size: sizeInBytes(fileDetails.fileB64Contents),
-      title: fileKey,
-    };
+
+    const attachment = buildAttachment(fileDetails, fileUrl, fileKey);
 
     if (docRef.date) attachment.creation = docRef.date;
     docRef.content = [{ attachment }];
-    const uploadParams = {
-      bucket: s3BucketName,
-      key: fileKey,
-      file: Buffer.from(fileDetails.fileB64Contents, "base64"),
-      ...(fileDetails.mimeType && { contentType: fileDetails.mimeType }),
-    };
+    const uploadParams = buildUploadParams(fileDetails, s3BucketName, fileKey);
 
     uploadDetails.push(uploadParams);
     docRefs.push(docRef);
   });
 
+  b64Attachments.organizers.map(organizerEntry => {
```

Comment:
this is the same code as above > 40% -- consider first mapping into the shared code requirements for each type (act and obsMedia) then have a single map to produce the arrays.

Reply:
Caveat: I don't see a date step, so maybe that's the difference?

Reply:
yea, there's no date, and the doc ref builder funcs are different (the component one has no type)